docs/src/modules/java/partials/query-syntax-reference.adoc

```diff
@@ -194,4 +251,7 @@ If no explicit ordering is specified in a view query, results will be returned i
 SELECT * FROM customers WHERE name = :name AND age > :min_age ORDER BY age DESC
 ----
 
-NOTE: Some orderings may be rejected, if the view index cannot be efficiently ordered. Generally, to order by a column it should also appear in the `WHERE` conditions.
\ No newline at end of file
+NOTE: Some orderings may be rejected, if the view index cannot be efficiently ordered. Generally, to order by a column it should also appear in the `WHERE` conditions.
```

Comment:
Tangent. I understand you didn't write this. Do you know what it means `efficiently ordered` and why `Generally` to order by a column, it should also appear in the `WHERE` condition?

Reply:
As for the first part, one example seems to be when streaming updates, but could also when trying to ORDER BY columns that are not indexed (see Indexes.scala#483 on the proxy).. For the second part, I think it's just saying that we probably don't want to sort every single row in the table, so having some sort of selection of rows is generally advisable.

Reply:
Thanks for the pointer 👍 👍  . I would advise rewriting this `Note` to let the user know what are those validations/limitations. 
Regarding the second I think the word `generally` implies that in some cases you can order by column even though it doesn't appear in the WHERE condition. [Indexes.scala#486](https://github.com/lightbend/kalix-proxy/blob/f3b42718793077b7033dcad8269db3a414bf8363/proxy/core/src/main/scala/kalix/proxy/views/Indexes.scala#L486) seems to indicate that is never the case.

Reply:
> I would advise rewriting this `Note` to let the user know what are those validations/limitations.

Let me see if I can do something about that..

> Regarding the second I think the word generally implies that in some cases you can order by column even if it doesn't appear in the WHERE condition. [Indexes.scala#486](https://github.com/lightbend/kalix-proxy/blob/f3b42718793077b7033dcad8269db3a414bf8363/proxy/core/src/main/scala/kalix/proxy/views/Indexes.scala#L486) seems to indicate that is never the case.

That code indeed seems to indicate such restriction.. however, yesterday I happen to have tried a query such `@Query("SELECT * FROM customers ORDER BY createdAt")` and it works.. I've just retried it now to make sure and is accepted. So maybe there's more to that code than what it looks at first sight.

Reply:
Nice catch! `createdAt` is a field that belongs to customer? or we are creating it?

Reply:
In that case it was part of a `CustomerSummary`, the view transformed type.

Reply:
Yes, there's more to it than always requiring these columns to be in the WHERE clause. Extra sort columns will also be indexed if possible — when it's possible to create an index in the expected order.

https://github.com/lightbend/kalix-proxy/blob/f3b42718793077b7033dcad8269db3a414bf8363/proxy/core/src/main/scala/kalix/proxy/views/Indexes.scala#L121

Reply:
We could expand the note to try explaining this in more detail. Or we could suggest people contact us if they run into the restrictions — as we can enable extra sort operations for some customers.

So by default, it only allows ORDER BY columns where an extra sort operation isn't needed — the index scan will return already ordered results, based on the indexing. So that's the 'efficiently' part. But if possible, it will add the sort columns to the index to create the expected order — it can do this when the columns indexed for the WHERE conditions are all exact matches (same for when there are no WHERE conditions).